services/faucet/src/modules/faucet/utils/utils.ts

```diff
@@ -0,0 +1,34 @@
+import { RequestEntity } from '../entity';
+
+export const filterRequestsOfThisWeek = (
+  requests: RequestEntity[]
+): RequestEntity[] => {
+  const monthDay = new Date().getDate();
+  const weekDay = new Date().getDay();
+  const daysToSunday = 7 - weekDay;
+  const daysFromSunday = weekDay;
+
+  const setDateToMidnight = (date: Date) => {
+    date.setHours(0);
+    date.setMinutes(0);
+    date.setSeconds(0);
+    date.setMilliseconds(0);
+  };
+  const maxDate = new Date();
+  maxDate.setDate(monthDay + daysToSunday);
+  setDateToMidnight(maxDate);
+  const minDate = new Date();
+  minDate.setDate(monthDay - daysFromSunday);
+  setDateToMidnight(minDate);
+
+  return requests.filter((req) => {
+    if (
+      req.createdAt.getTime() < maxDate.getTime() &&
```

Comment:
When comparing the dates here, we need to make sure that both the request date and this date are in UTC.